tests/framework-tests/core/js/ModelTransformationTests.js

```diff
@@ -2652,10 +2750,274 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+        },
+        expected: {
+            value: "dog"
+        },
+        fullyinvertible: true
+    }, {
+        message: "dereference() should return the value when the \"array\" is arrayable and match the given index.",
+        transform: {
+            value: {
+                transform: {
+                    type: "fluid.transforms.dereference",
+                    array: "sheep",
+                    inputPath: "element"
+                }
+            }
+        },
+        invertedRules: {
+            transform: [{
+                type: "fluid.transforms.indexOf",
+                array: "sheep",
+                outputPath: "element",
+                inputPath: "value"
+            }]
+        },
+        method: "assertDeepEq",
+        model: {
+            element: 0
+        },
+        expected: {
+            value: "sheep"
+        },
+        fullyinvertible: true
+    }, {
+        message: "dereference() should take the offset value into consideration.",
+        transform: {
+            value: {
+                transform: {
+                    type: "fluid.transforms.dereference",
+                    array: ["sheep", "dog"],
+                    offset: 3,
+                    inputPath: "element"
+                }
+            }
+        },
+        invertedRules: {
+            transform: [{
+                type: "fluid.transforms.indexOf",
+                array: ["sheep", "dog"],
+                offset: 3,
+                outputPath: "element",
+                inputPath: "value"
+            }]
+        },
+        method: "assertDeepEq",
+        model: {
+            element: 4
+        },
+        expected: {
+            value: "dog"
+        },
+        fullyinvertible: true
+    }, {
+        message: "dereference() should take the offset value into consideration if the offset value can be converted to a number.",
+        transform: {
+            value: {
+                transform: {
+                    type: "fluid.transforms.dereference",
+                    array: ["sheep", "dog"],
+                    offset: "1",
+                    inputPath: "element"
+                }
+            }
+        },
+        invertedRules: {
+            transform: [{
+                type: "fluid.transforms.indexOf",
+                array: ["sheep", "dog"],
+                offset: "1",
+                outputPath: "element",
+                inputPath: "value"
+            }]
+        },
+        method: "assertDeepEq",
+        model: {
+            element: 2
+        },
+        expected: {
+            value: "dog"
+        },
+        fullyinvertible: true
+    }, {
+        message: "dereference() should ignore the offset when the offset value cannot be converted a number.",
+        transform: {
+            value: {
+                transform: {
+                    type: "fluid.transforms.dereference",
+                    array: ["sheep", "dog"],
+                    offset: "a",
```

Comment:
This should actually be a hard error rather than ignored. If the offset converts to NaN or a non-integer, the framework should throw.